rotkehlchen/chain/evm/decoding/pendle/utils.py

```diff
@@ -0,0 +1,104 @@
+
+from .constants import CPT_PENDLE
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_pendle_markets(chain: ChainID) -> None:
+    """Query Pendle markets for a given chain and cache their addresses."""
+    all_markets, all_sy_tokens = [], []
+    try:
+        for market_type in ['active', 'inactive']:  # query active and inactive markets
+            response_data = request_get_dict(url=f'https://api-v2.pendle.finance/core/v1/{chain.value}/markets/{market_type}')
+            for market in response_data['markets']:
+                try:
+                    all_markets.append(deserialize_evm_address(market['address']))
+                    all_sy_tokens.append(deserialize_evm_address(market['sy'].split('-')[1]))
+                except (KeyError, ValueError) as e:
+                    log.warning(f'Skipping pendle market entry {market} due to {e!s}')
+                    continue
+
+    except (RemoteError, UnableToDecryptRemoteData, TypeError, KeyError) as e:
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+        log.error(f'Failed to retrieve Pendle markets for chain {chain} due to {msg}')
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.PENDLE_POOLS, str(chain.value)),
+            values=all_markets,
+        )
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.PENDLE_SY_TOKENS, str(chain.value)),
+            values=all_sy_tokens,
+        )
+
+
+def query_pendle_price(token: 'EvmToken') -> Price:
+    """Query USD price for a given Pendle PT/YT/SY/LP token"""
+    chain = token.chain_id
+    try:
+        result = request_get_dict(f'https://api-v2.pendle.finance/core/v1/{chain.value}/assets/prices?addresses={token.evm_address}')
+        return deserialize_price(result['prices'][to_normalized_address(token.evm_address)])
+    except (RemoteError, UnableToDecryptRemoteData, DeserializationError, TypeError, ValueError, KeyError) as e:  # noqa: E501
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+        log.error(f'Unable to query usd price of pendle wrapped token {token} due to {msg}')
+        return ZERO_PRICE
+
+
+def query_pendle_yield_tokens(evm_inquirer: 'EvmNodeInquirer') -> None:
+    """Queries all Pendle yield tokens(LP, SY, PT, YT) and saves them to allow price queries."""
+    all_tokens: set[ChecksumEvmAddress] = set()
+    try:
+        all_tokens.update(deserialize_evm_address(entry['address']) for entry in request_get_dict(f'https://api-v2.pendle.finance/core/v3/{evm_inquirer.chain_id.value}/assets/all'))
```

Comment:
Use serialize where you use value since it is easier to change it later

```suggestion
        all_tokens.update(deserialize_evm_address(entry['address']) for entry in request_get_dict(f'https://api-v2.pendle.finance/core/v3/{evm_inquirer.chain_id.serialize()}/assets/all'))
```